include/integration/simple_MC_integration.hpp

```diff
@@ -0,0 +1,194 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2021 Vissarion Fisikopoulos
+// Copyright (c) 2018-2021 Apostolos Chalkis
+
+// Contributed and/or modified by Suraj Choubey, as part of Google Summer of Code 2021 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#ifndef SIMPLE_MC_INTEGRATION_HPP
+#define SIMPLE_MC_INTEGRATION_HPP
+
+#include <iostream>
+#include <fstream>
+#include <random>
+#include <vector>
+#include "convex_bodies/hpolytope.h"
+#include "Eigen/Eigen"
+#include "generators/known_polytope_generators.h"
+#include "boost_random_number_generator.hpp"
+#include "cartesian_geom/cartesian_kernel.h"
+#include "random_walks/random_walks.hpp"
+#include "volume/volume_sequence_of_balls.hpp"
+#include "volume/volume_cooling_gaussians.hpp"
+#include "volume/volume_cooling_balls.hpp"
+#include "misc.h"
+
+typedef double NT;
+typedef Cartesian<NT> Kernel;
+typedef typename Kernel::Point Point;
+typedef std::vector<Point> Points;
+typedef HPolytope<Point> HPOLYTOPE;
+typedef boost::mt19937 RNGType;
+typedef BoostRandomNumberGenerator<RNGType, NT> RandomNumberGenerator;
+typedef typename HPolytope<Point>::MT MT;
+typedef typename HPolytope<Point>::VT VT; 
+
+typedef const unsigned int Uint;  // positive constant value for no of samples & dimensions
+enum volType { CB , CG , SOB }; // Volume type for polytope
+
+// To check if two n-dimensional points ensure valid limits in integration 
+bool validLimits(Point LL, Point UL){
+    if( UL.dimension() == LL.dimension() ) {
+        for ( int i = 0 ; i< LL.dimension() ; i++){
+            if( LL[i] > UL[i]){
+            std::cout << "Invalid integration limits\n";
+            return false;
+            }
+        }
+        return true;
+    }else{
+        std::cout << "Invalid integration limits\n";
+        return false;
+    }
+}
+
+// Hyper-Rectangle volume calculator in n-dimensions
+NT hyperRectVolume(Point LL, Point UL){
+    NT product=1;
+    if(validLimits(LL,UL)){
+        for(int i=0; i<LL.dimension(); ++i){
+            product = product * ( UL[i] - LL[i] );
+        }
+        return product;
+    }
+    else return 0;
+}
+
+// To sample a point between two n-dimensional points using inbuilt random sampling
+// Point samplerBWLimits(Point LL, Point UL){
+//     Point sample_point(LL.dimension());
+//     for(int i=0; i<LL.dimension(); ++i){
+//         sample_point.set_coord(i , LL[i] + (NT)(rand()) / ((NT)(RAND_MAX/(UL[i] - LL[i]))) );
+//     }
+//     return sample_point;
+// }
+
+// Simple MC Integration over Hyper-Rectangles
+template
+<
+    typename WalkType=BallWalk,
+    typename Functor
+>
+void simple_mc_integrate(Functor Fx, Uint N ,Point LL, Point UL, int walk_length=10, NT e=0.1){
```

Comment:
Should this function take a generic polytope as an argument? 
You can create an extra function which is more user-friendly: ie it takes Point LL and Point UL and then calls this function. 

Moreover, please check the spacing in the code which is not consistent.

Reply:
Well not actually `simple_mc_integrate()` is designed to take integration limits as `Points`. (This is come up with thought to ease the user)

`simple_mc_polytope_integrate()` is designed to take generic polytope as argument.